src/runtime/hashmap.h

```diff
@@ -6,6 +6,14 @@
 #include "printer.h"
 #include "scoped_mutex_lock.h"
 
+#ifndef hashmap_malloc
+#define hashmap_malloc(size)    halide_malloc(NULL, size)
+#endif//hashmap_malloc
```

Comment:
Two spaces between #endif and //

Reply:
halide_malloc and halide_free take a user_context; clients that override these may fail if you pass in null. You need to add user_context as arguments to the hashmap_malloc/free macros and pass them everywhere. (If that isn't possible, you can't use halide_malloc.)

Reply:
I am now preserving the `user_context` around it.

However, I should point that `cache.cpp` (which is from I based the `hashmap.h` code), `halide_malloc()` and `halide_free()` are used without `user_context`.
(I think `cache.cpp` is @zvookin's domain)